xonsh/history/json.py

```diff
@@ -635,6 +398,58 @@ def flush(self, at_exit=False):
         self.buffer.clear()
         return hf
 
+    def do_gc(self, wait_for_shell, size):
+        self.gc = JsonHistoryGC(wait_for_shell=False, size=size)
+        return self.gc
+
+    def session_items(self, **kwargs):
+        return iter(self)
+
+    # TODO: merge methods all_items() and items() to one
+    def all_items(self, **kwargs):
+        """
+        Returns all history as found in XONSH_DATA_DIR.
+
+        return format: (cmd, start_time, index)
+        """
+        ind = 0
+        for f in _get_history_files():
+            try:
+                json_file = xlj.LazyJSON(f, reopen=False)
+            except ValueError:
+                # Invalid json file
+                continue
+            commands = json_file.load()['cmds']
+            for c in commands:
+                yield (c['inp'].rstrip(), c['ts'][0], ind)
+                ind += 1
+
+    # TODO: merge methods all_items() and items() to one
```

Comment:
One of them were used to feed history items to readline (when do history search etc); one is for `history show all`, I think we can merge them into one. I will work on them.